component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java

```diff
@@ -480,6 +482,98 @@ public List<String> getUserIdsActivities(Identity owner, long offset, long limit
     return activityDAO.getUserIdsActivities(owner, offset, limit);
   }
 
+
+  @Override
+  public List<ExoSocialActivity> getActivitiesByFilter(Identity viewerIdentity,
+                                                       ActivityFilter activityFilter,
+                                                       long offset,
+                                                       long limit) {
+    List<String> spaceIdentityIds = null;
+    switch (activityFilter.getStreamType()) {
+    case USER_STREAM:
+      activityFilter.setUserId(viewerIdentity.getId());
+      break;
+    case FAVORITE_SPACES_STREAM:
+      spaceIdentityIds = spaceStorage.getFavoriteSpaceIdentityIds(viewerIdentity.getRemoteId(), new SpaceFilter(), 0, -1);
+      if (CollectionUtils.isEmpty(spaceIdentityIds)) {
+        return Collections.emptyList();
+      }
+      break;
+    case MANAGE_SPACES_STREAM:
+      spaceIdentityIds = spaceStorage.getSpaceIdentityIdsByUserRole(viewerIdentity.getId(),
```

Comment:
Currently the method `spaceStorage.getSpaceIdentityIdsByUserRole` (storage layer) receives the Identity ID and then the DAO layer (spaceMemberDAO.getSpaceIdentityIdsByUserRole) needs the username (remoteId). IMO it would be better to change the storage layer to receive the `username` instead of identityId to optimize useless database operation.

N.B: make sure that this method isn't used outside social project after changing the parameter type.